### ISSUE TITLE
Can achieve accuracy of plus minus 2-3cm by 1m distance?

### ISSUE THREAD

**Issue by astronaut71** [chensong1995/HybridPose#34]
H

I have to estimate the 6D Pose of regular shape objects(like construction blocks ) in the construction field. So it is for the grasping in pick and place task. Roughly the camera will be positioned around 1m of the object (the construction block) and I need to get the accurate position and orientation of the top blocks, The blocks can be on top of each other. Would this approach work on real-time (around 20-30fps) and accuracy of 2-3cm by 1m distance?

I have Cuda 10 and GPU card as well.

Thanks

**Comment by chensong1995** [comment]
Hi astronaut71, 

Thanks for your question! The accuracy on a new dataset will largely depend on the accessibility of good-quality training data. Without much information about the specific setting of your task, I am unable to provide accuracy guarantee. 

The overall pipeline should be able to run in real-time, provided that the resolution of input image is not super high (~1000*1000 should be fine). Please note that you may need to write a function that perform pose inference on multiple frames in parallel yourself. The function is mostly identical to `generate_data()` in `trainer.py`. The only difference is that multiple calls to the pose regressor in C++ should now be made in parallel. 

I hope this helps.

**Comment by astronaut71** [comment]
Thanks for the quick answer. And does this work with real data? What more information about the specific setting of my task do you requered?

**Comment by chensong1995** [comment]
Real-world application is harder than experimental tests. I think if you can gather a training dataset which has roughly the same size as linemod (~1000 examples), it will be worth trying it out.

**Comment by astronaut71** [comment]
so I need around 1000 images for training? And works on real-time, let say 20-30fps?

**Comment by chensong1995** [comment]
They are both rough estimates. The exact number of training examples depends on the particular problem, and also how well the training goes with a small training set. You may need to collect more data if the small training set doesn't work.

**Comment by astronaut71** [comment]
ok. Sorry, Last question. Is it comaprinf kknown 3D CAD model od the object with corresponding 2D point in the Image more accurate? Or any other approach?

**Comment by chensong1995** [comment]
I'm not sure if I understand you correctly. There are different ways of aligning the 3d model to 2d information. Using keypoints is a popular way, but they're others. In our work, we use edge vectors and symmetry correspondence in addition to keypoints for pose alignment. Other relevant work such as DPOD uses "correspondence mapping" and also achieves good performance.

**Comment by astronaut71** [comment]
In my case, the object doesn't have any textures. So it's a non-textured object. I was thinking of aligning the 3d model to 2d information, using e.g. contour of the object, harris corners detection. Would be that a good way? What is the difference between your way of aligning using edge vectors- symmetry correspondence and using local features (contours, harris corners..)?
